### DISCUSSION TITLE
Dynamic PostGIS queries to render vector tiles

### DISCUSSION THREAD

**Discussion by zmatez** [maplibre/martin#1525] [General]
Hello, I'm wondering if I can pass to vector URL specific query I want to execute and get tile based on it. I don't want to create functions for each thing I want to render, also some queries are needed to be generated basing on user input and permissions.
For security user wouldn't be quering db by itself, but by broker as Java server (having auth and external parameters), to load martin's vector tile and serve it to end user.
Otherwise I'd have to create really fancy functions and use parametrized queries that I've seen available in Martin (but still use Java as a broker)
Thanks

**Comment by 0x-a6** [comment]
If I understand your stack, you are wanting to have martin tiles served through some Java application based on tiles requested to the Java application? 
If that's the case: The URL is just the base URL domain name + /{z}/{z}/{y} so you can just request the same through your Java application with an auth middleware for example. You can pass the parameters in the URL to the martin instance via and advanced spring web client for example: https://docs.spring.io/spring-framework/reference/web/webflux-webclient.html
Then the mapping application or external functions can just request the same tiles and have Java can pass through the tiles after any work you want your Java application to apply. Just beware that this would slow down the tile requests.

**Reply by zmatez** [reply]
Yeah but I'm asking about certain martin usage - is it possible to request tiles basing on postgis SQL query on the go? Meaning SQL generated in that Java application and requested to Martin? Instead creating postgres functions for each query which is not ideal in my case
